promql/value.go

```diff
@@ -264,10 +294,11 @@ func (m Matrix) String() string {
 }
 
 // TotalSamples returns the total number of samples in the series within a matrix.
+// It takes into account the number of samples in the histograms using the histogramSize method.
```

Comment:
Maybe: "Float samples have a weight of 1 in this number, while histogram samples have a higher weight according to their size compared with the size of a float sample. See HPoint.histogramSize for details."